src/main/java/edu/unc/lib/boxc/migration/cdm/services/ListProjectsService.java

```diff
@@ -0,0 +1,148 @@
+        if (Files.notExists(directory)) {
+            throw new InvalidProjectStateException("Path " + directory + " does not exist");
+        }
+
+        // create JSON
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.registerModule(new JavaTimeModule());
+        mapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+        ArrayNode arrayNode = mapper.createArrayNode();
+
+        for (File file : directory.toFile().listFiles()) {
+            if (file.isDirectory()) {
+                MigrationProject project = initializeProject(file.toPath());
+
+                Path projectPath = directory.toAbsolutePath();
+                String projectStatus = status(project);
+                String allowedActions = allowedActions(project);
+                JsonNode projectProperties = mapper.readTree(project.getProjectPropertiesPath().toFile());
+
+                // add project info to JSON
+                ObjectNode objectNode = mapper.createObjectNode();
+                objectNode.put(PROJECT_PATH, projectPath.toString());
+                objectNode.put(STATUS, projectStatus);
+                objectNode.put(ALLOWED_ACTIONS, allowedActions);
+                objectNode.set("projectProperties", projectProperties);
+                arrayNode.add(objectNode);
+            }
+        }
+        log.debug(arrayNode.toString());
+
+        return arrayNode;
+    }
+
+    /**
+     * Condensed representation of the phase a project is in
+     * possible values: initialized, indexed, sources_mapped, sips_generated, ingested, archived
+     * @return project status
+     */
+    private String status(MigrationProject project) {
+        String status = null;
+
+//        TODO: add archived state
+//        if () {
+//            status = "archived";
+//        }
+        if (!project.getProjectProperties().getSipsSubmitted().isEmpty()) {
+            status = "ingested";
+        } else if (project.getProjectProperties().getSipsGeneratedDate() != null) {
+            status = "sips_generated";
+        } else if (project.getProjectProperties().getSourceFilesUpdatedDate() != null) {
+            status = "sources_mapped";
+        } else if (project.getProjectProperties().getIndexedDate() != null) {
+            status = "indexed";
+        } else if (Files.exists(project.getProjectPropertiesPath())) {
+            status = "initialized";
+        }
+
+        return status;
+    }
+
+    /**
+     * Tell other applications (boxc) what processes are supported by the project
+     * crop_color_bars: populated if source files are mapped and if project contains any images (based off source file extensions)
+     * @return allowed_actions
+     */
+    private String allowedActions(MigrationProject project) throws Exception {
+        String allowedActions = null;
+        List<String> imageFormats = Arrays.asList("tif", "jpeg", "png", "gif", "pict", "bmp",
```

Comment:
since these are file extensions, we'll want to also include "tiff" and "jpg" since those occur commonly in our data. Also, a Set/HashSet is probably more appropriate for this since it provides faster "contains". And we should turn this into a constant.